lib/fs.js

```diff
@@ -63,6 +66,16 @@ const isWindows = process.platform === 'win32';
 const DEBUG = process.env.NODE_DEBUG && /fs/.test(process.env.NODE_DEBUG);
 const errnoException = util._errnoException;
 
+const doFlaggedDeprecation = noDeprecation ?
+  function() {} : showFlaggedDeprecation;
```

Comment:
This part is unnecessary. The `emitWarning` function will handle the `noDeprecation` configuration property internally.

Reply:
This was suggested by one of the reviewers, I am just here to learn how nodejs works internally. If you think `emitWarning` is sufficient, i can make the change.